### PR TITLE
Fixes hulk arms breaking when punching non walls

### LINKED ISSUE #52176: Hulks break arms no matter what they punch
#51389 was supposed to make punching walls as a hulk do damage to your arms and break them if you punched too many, but it did that very poorly. Hulk punching code deals guaranteed wounds based on how much damage your arm has when you punch, and it does this no matter how your arm got damaged or what you're punching, even if you get shot in the arm in a fight and then punch your attacker. 

I assume this is unintended behavior because the PR doesn't mention anything about that and it's horribly inconsistent with how wounds is supposed to work, which drags me into it because bone wounds are intended to be rare and time consuming to fix.

### PR DESCRIPTION
**Pull request by zxaber**
## About The Pull Request
- Adds a new proc to handle hulk arm breaking based off damage, appropriately called `break_an_arm()`. This is intended to be called by the atom being punched, during the `attack_hulk()` chain. This fixes hulk arm breaking when punching unintended things (like mobs). Additionally, a call to `break_an_arm()` was *intentionally* left out of machine code, meaning that while punching an airlock or prothlathe will deal the same slight damage, it cannot trigger an arm break (even at max arm damage).

- Fixes #52176

## Why It's Good For The Game

- Fixes the bug listed above.

- Since we can now deal penalty damage without breaking the arm, I can lower the machine punching penalty to be still annoying without actually carrying the same effects that wall punching does. In effect, you can now break as many airlocks as you wish (but if your arm is at max damage, it'll break the first time you successfuly smash a wall).

## Changelog
:cl:
fix: Hulk arms no longer break with punching things unintended to break hulk arms
balance: Hulk arms no longer break when punching machines (but still take a small bit of recoil damage).
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
speedmerge

File: code/game/turfs/closed/walls.dm [hunk 1]
```diff
@@ -163,12 +163,18 @@
...
   *smashed walls per arm. Also, we use CANT_WOUND here because wounds are random. Wounds are applied
-  *by hulk code based on arm damage and checked after an attack completes.
+  *by hulk code based on arm damage and checked when we call break_an_arm().
   *Arguments:
   **arg1 is the arm to deal damage to.
+  **arg2 is the hulk
  */
-/turf/closed/wall/proc/hulk_recoil(obj/item/bodypart/arm)
+/turf/closed/wall/proc/hulk_recoil(obj/item/bodypart/arm, mob/living/carbon/human/hulkman)
 	arm.receive_damage(brute = 20, blocked = 0, wound_bonus = CANT_WOUND)
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/turfs/closed/walls.dm:172]: if you add something like

```DM
/turf/closed/wall/proc/hulk_recoil(obj/item/bodypart/arm, mob/living/carbon/human/hulkman, inherit_damage = TRUE)
    if(inherit_damage)
	    arm.receive_damage(brute = 20, blocked = 0, wound_bonus = CANT_WOUND)
```
You can avoid the copypasta, use inherit, and just set `inherit_damage` to FALSE

**Comment by zxaber** [comment]
Github is having a stroke

File: code/game/turfs/closed/walls.dm (hunk 1 above)
Comment (zxaber) [reply code/game/turfs/closed/walls.dm]: Done, used `var/damage = 20` so that I could set it to both 41 and 0 depending on the wall type.

**Pull request merged**